### ISSUE TITLE
[virtual paging] Infinite loop on virtual paging issue

### ISSUE THREAD

**Issue by cantacell** [ag-grid/ag-grid#554]
If you modify the pageSize to 5 in your virtual paging example (http://www.ag-grid.com/angular-grid-virtual-paging/virtualPaging.html) you'll have an infinite loop on getRows function.
Thanks

**Comment by cantacell** [comment]
Up!
If the tutorial doesn't work, this grid is USELESS!
Thanks!

**Comment by ceolter** [comment]
```
If you modify the pageSize to 5
```
-> don't modify the pageSize to 5!!

what's wrong with the tutorial?

```
this grid is USELESS!
```
-> go use anther grid then!

**Comment by cantacell** [comment]
Why I cannot modify the page size?? 
Previously I said " if the tutorial doesn't work THEN the grid is useless"....

**Comment by ceolter** [comment]
re page size, it comes down to maths.

if the pageSize is 5, and the maxPagesInCache is 2, then the total number of rows in the cache is 10 rows (5 x 2). so if your grid can display 40 rows on the screen (pretty normal) then your cache isn't big enough to hold one displayed set of data, thus the cache keeps trashing trying to load the rows.

the grid also renders some rows above and below the screen as a buffer (default is 20 each way) which you need to consider also.

so in summary, if you reduce the page size, you need to ensure you still have a cache size (pageSize x pagesInCache) big enough to hold enough rows to render your grid.

**Comment by ceolter** [comment]
what do you mean by the tutorial doesn't work? what is wrong with it?

**Comment by cantacell** [comment]
I tried the tutorial and sometimes I encountered an infinite-loop. Then I discovered that if set the pagesize to 5 the infinite-loop always occurs.
Tomorrow I will try to change the maxPagesInCache and I will send you a feedback.
Best regards,
Francesco

**Comment by ceolter** [comment]
```
I tried the tutorial and sometimes I encountered an infinite-loop
```
the tutorial doesn't use server side paging, so the infinite loop must be something else???

**Comment by cantacell** [comment]
I tried again the tutorial.
In my browser window I can see 36 rows at time, so if I put pageSize:36 I must put maxPagesInCache: 3 else there is a bad render rows. So the developer must know how many rows can be visible in the grid at time, BUT if you try to zoom out the page you will encounter in the same issue.

![bug_ag-grid](https://cloud.githubusercontent.com/assets/15688388/12425604/e6f13b0c-bed5-11e5-9aa1-3afd1a6fe465.PNG)

**Comment by ceolter** [comment]
i wasn't programming with 'zoom out' in mind, so yes, virtual pagination breaks, if you zoom out far enough.

**Comment by cantacell** [comment]
Ok, so in general the developer must know how many rows are visible before the grid is rendered...
Thanks for the support.

**Comment by ceolter** [comment]
yup, that's so. i wasn't expecting the grid to grow in 'number of rows' at runtime.

**Comment by ceolter** [comment]
tracked by https://github.com/ceolter/ag-grid/issues/554
